tests/token_provision.robot

```diff
@@ -0,0 +1,32 @@
+*** Settings ***
+Library     SSHLibrary    timeout=90 seconds
+Library     Process
+Library     OperatingSystem
+Library     String
+Library     RequestsLibrary
+Library     Collections
+
+Suite Setup       Run Keyword    Open Server
+Suite Teardown    Run Keyword    SSHLibrary.Close All Connections
+
+Resource    ../variables.robot
+Resource    ../keywords.robot
+
+*** Test Cases ***
+
+TOP001.001 Token provisioning
+    [Documentation]    Checks that API command admin/token_provision works
+    ...                correctly.
+    [Teardown]    Run Keyword If Test Failed    Set Suite Variable    ${prev_test_status}    FAIL
+    Run    python3 provisioning/cc_cbor.py provisioning/chain.pem provisioning/tmp/fobnail.cbor
+    Run    coap-client -t application/cbor -m post -f provisioning/tmp/fobnail.cbor coap://${fobnail_ip}/api/v1/admin/token_provision > provisioning/tmp/fobnail.csr
+    SSHLibrary.Read Until    Certificate chain loaded
+    SSHLibrary.Read Until    Generating new
```

Comment:
We should check CoAP response, not Fobnail logs. We want to check whether Fobnail behaves as described [here](https://fobnail.3mdeb.com/fobnail-api/). This document defines Fobnail requests and responses, not what messages Fobnail prints. After `Certificate Chain loaded` message many things can happen (see [this link](https://github.com/fobnail/fobnail/blob/b9814d3746475497678f7c459620407abc9c073b/src/server/token_provisioning/mod.rs#L196)), e.g. Fobnail may fail to produce CSR or may fail to send the response. In that case we would have a false-positive test pass.

Reply:
applied